src/HospitalLibrary/Patients/Validators/PatientRegistrationValidationService.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using HospitalLibrary.Patients;
+using HospitalLibrary.Shared.Exceptions;
+using HospitalLibrary.Shared.Interfaces;
+
+namespace HospitalLibrary.Shared.Validators
+{
+    public class PatientRegistrationValidationService: IPatientRegistrationValidationService
+    {
+        private IUnitOfWork _unitOfWork;
+        
+        public PatientRegistrationValidationService(IUnitOfWork unitOfWork)
+        {
+            _unitOfWork = unitOfWork;
+        }
+
+        public async Task ValidatePatient(Patient patient)
+        {
+            if (await _unitOfWork.PatientRepository.GetOneByUid(patient.Uid) != null)
+                throw new BadRequestException("Uid is already taken");
```

Comment:
If you do validation like this, then you don't need to capture exception inside of service.